toc/rfc/rfc-draft-generic-per-route-features.md

```diff
@@ -0,0 +1,157 @@
+# Meta
+[meta]: #meta
+- Name: Generic Per-Route Features
+- Start Date: 2024-01-08
+- Author(s): @maxmoehl, @domdom82
+- Status: Draft <!-- Acceptable values: Draft, Approved, On Hold, Superseded -->
+- RFC Pull Request: (fill in with PR link after you submit it)
+
+## Summary
+
+## Problem
+
+Implementing per-route features in Cloud Foundry is a tedious process because it requires changes to
+multiple components in multiple working groups. Most components will only need to be made aware of
+new fields without adding any significant logic but the overhead of making the change still exists.
+
+On the other side are the components that actually need to be aware of the change. Mostly the
+routing components (gorouter, maybe route-emitter / -registrar), the API (cloud controller) and
+UIs (CLI).
+
+## Proposal
+
+To reduce the overall overhead of implementing new per-route features we propose to add a generic
+key-value map to the definition of a route across all components. All components handling the map
+SHOULD ignore the contents of the map except gorouter, cloud controller and cf CLI which MUST handle
+the map and its values accordingly. Other components MAY limit the size of the map or size of keys /
+values for technical reasons.
+
+The following constraints apply (types are as specified in [RFC 8259](https://rfc-editor.org/rfc/rfc8259)):
+* The map MUST be representable as an object.
+* The map MUST only use strings as keys.
+* The map MUST only use numbers, strings and the literal values `true` and `false` as values.
+
+### Required Changes
+
+#### CLI
+
+There MAY be either individual command line flags or a single flag which supports generic key-value
+pairs for the `map-route` sub-command.
```

Comment:
I would go for a single flag which supports generic key-value pairs in cli. This allows to keep it generic. CC and gorouter would be the only participants that need to have dedicated knowledge about the properties.

Similar to the -c option for specifying service instance parameters in `cf create-service`.

Reply:
I've adjusted the wording accordingly.